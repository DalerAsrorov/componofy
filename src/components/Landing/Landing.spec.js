import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createMount, createShallow } from 'material-ui/test-utils';
import Tabs, { Tab } from 'material-ui/Tabs';
import Lainding from './';

Enzyme.configure({ adapter: new Adapter() });

// Required props for Material UI components context
const requiredProps = {
  onAuth: jest.fn(),
  title: 'Login',
  subTitle: 'Some login',
};

describe('When Nav component is initialized', () => {
  let mount, shallow;

  beforeEach(() => {
    mount = createMount();
    shallow = createShallow();
  });

  it('Landing has correct props', () => {
    let wrapper = shallow(<Lainding {...requiredProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should have a correct title', () => {
    let wrapper = mount(<Lainding {...requiredProps} />);

    expect(wrapper.prop('title')).toBe(requiredProps.title);
  });
});
