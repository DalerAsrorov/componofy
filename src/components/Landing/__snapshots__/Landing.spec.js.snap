// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Nav component is initialized Landing has correct props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Landing)
    onAuth={[Function]}
    subTitle="Some login"
    title="Login"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "authBtn": "Landing-authBtn-4",
        "demoLink": "Landing-demoLink-1",
        "icon": "Landing-icon-5",
        "iconText": "Landing-iconText-6",
        "root": "Landing-root-2",
        "subheader": "Landing-subheader-3",
      },
      "onAuth": [Function],
      "subTitle": "Some login",
      "title": "Login",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "authBtn": "Landing-authBtn-4",
          "demoLink": "Landing-demoLink-1",
          "icon": "Landing-icon-5",
          "iconText": "Landing-iconText-6",
          "root": "Landing-root-2",
          "subheader": "Landing-subheader-3",
        },
        "onAuth": [Function],
        "subTitle": "Some login",
        "title": "Login",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
